swift/apple/FirezoneNetworkExtension/PacketTunnelProvider.swift

```diff
@@ -93,23 +117,30 @@ class PacketTunnelProvider: NEPacketTunnelProvider {
       let internetResourceEnabled =
         enabled != nil ? enabled == "true" : (tunnelConfiguration?.internetResourceEnabled ?? false)
 
-      let adapter = Adapter(
-        apiURL: apiURL,
-        token: token,
-        id: id,
-        logFilter: logFilter,
+      // Create and start the adapter
+      let adapter = AdapterUniFfi(packetTunnelProvider: self)
+      self.adapter = adapter
+
+      try adapter.start(
+        apiUrl: apiURL,
+        token: token.description,
+        deviceId: id,
         accountSlug: accountSlug,
-        internetResourceEnabled: internetResourceEnabled,
-        packetTunnelProvider: self
+        logFilter: logFilter
       )
 
-      try adapter.start()
-
-      self.adapter = adapter
+      // Set initial internet resource state after adapter is started
+      do {
+        try adapter.setInternetResourceEnabled(internetResourceEnabled)
+        Log.log("Initial internet resource enabled set to: \(internetResourceEnabled)")
+      } catch {
+        Log.error(error)
+      }
```

Comment:
Hmmm perhaps we want to let the catch block just below this handle any possible failures here? I.e. be noisy if we fail to setInternetResourceEnabled?